src/reason-parser/reason_pprint_ast.ml

```diff
@@ -3273,6 +3274,38 @@ let printer = object(self:'self)
         else Some (self#formatJSXComponent (Longident.last loc.txt) l)
       else None
     )
+    | (Pexp_apply ({pexp_desc=Pexp_letmodule(_, ({pmod_desc=Pmod_apply(m1, m2)} as app), {pexp_desc=Pexp_ident loc})}, l), [], _jsx::_) -> (
+      (* TODO: Soon, we will allow the final argument to be an identifier which
+         represents the entire list. This would be written as
+         `<tag>...list</tag>`. If you imagine there being an implicit [] inside
+         the tag, then it would be consistent with array spread:
+         [...list] evaluates to the thing as list.
+      *)
+      let rec extract_apps args = function
+        | { pmod_desc = Pmod_apply (m1, {pmod_desc=Pmod_ident loc}) } ->
+          let arg = String.concat "." (Longident.flatten loc.txt) in
+          extract_apps (arg :: args) m1
+        | { pmod_desc=Pmod_ident loc } -> (String.concat "." (Longident.flatten loc.txt))::args in
+      let hasLabelledChildrenLiteral = List.exists (function
+        | (Labelled "children", _) -> true
+        | _ -> false
+      ) l in
+      let rec hasSingleNonLabelledUnitAndIsAtTheEnd l = match l with
+      | [] -> false
+      | (Nolabel, {pexp_desc = Pexp_construct ({txt = Lident "()"}, _)}) :: [] -> true
+      | (Nolabel, _) :: rest -> false
+      | _ :: rest -> hasSingleNonLabelledUnitAndIsAtTheEnd rest
+      in
+      if hasLabelledChildrenLiteral && hasSingleNonLabelledUnitAndIsAtTheEnd l then
+        if List.length (Longident.flatten loc.txt) > 1 then
+          if Longident.last loc.txt = "createElement" then
+            let ftor::args = extract_apps [] app in
```

Comment:
non-exhaustive pattern matching here too

perhaps the code can be refactored a bit